core/class/jMQTTCmd.class.php

```diff
@@ -310,25 +340,72 @@ public function postSave() {
                 // If it's topic changed
                 if ($this->_preSaveInformations['brokerStatusTopic'] != $this->getTopic()) {
                     // Just try to remove the previous status topic
-                    $eqLogic = $this->getEqLogic();
                     $eqLogic->publish($eqLogic->getName(), $this->_preSaveInformations['brokerStatusTopic'], '', 1, 1);
                 }
             }
         }
 
-        // For info commands, check that the topic is compatible with the subscription command
-        // of the related equipment
-        if ($this->getType() == 'info' && $this->getEqLogic()->getType() == jMQTT::TYP_EQPT && !$this->getEqLogic()->getCache(jMQTT::CACHE_IGNORE_TOPIC_MISMATCH, 0)) {
-            if (! $this->topicMatchesSubscription($this->getEqLogic()->getTopic())) {
-                $this->eventTopicMismatch();
+        // For Equipments
+        if ($eqLogic->getType() == jMQTT::TYP_EQPT) {
+            // For info commands, check that the topic is compatible with the subscription command
+            if ($this->getType() == 'info' && !$eqLogic->getCache(jMQTT::CACHE_IGNORE_TOPIC_MISMATCH, 0)) {
+                if (! $this->topicMatchesSubscription($eqLogic->getTopic())) {
+                    $this->eventTopicMismatch();
+                }
             }
+			$this->listenerUpdate();
         }
     }
 
+// Listener for auto_publish action command
+	public function listenerUpdate() {
+		$cmds = array();
+		if ($this->getEqLogic()->getIsEnable() && $this->getType() == 'action' && $this->getConfiguration('auto_publish', 0)) {
+			preg_match_all("/#([0-9]*)#/", $this->getConfiguration('request', ''), $matches);
+			$cmds = array_unique($matches[1]);
+		}
+		$listener = listener::byId($this->getCache('listener', null));
+		if (count($cmds) > 0) { // We need a listener
+			if (!is_object($listener))
+				$listener = new listener();
+			$listener->setClass(__CLASS__);
+			$listener->setFunction('listenerAction');
+			$listener->emptyEvent();
+			foreach ($cmds as $cmd_id) {
+				$cmd = cmd::byId($cmd_id);
+				if (is_object($cmd) && $cmd->getType() == 'info')
+					$listener->addEvent($cmd_id);
+			}
+			$listener->setOption('cmd', $this->getId());
+			$listener->setOption('background', false);
+			$listener->save();
+			$this->setCache('listener', $listener->getId());
+			log::add('jMQTT', 'debug', 'Installed Listener on #'.$this->getHumanName().'#');
+		} else { // We don't want a listener
+			if (is_object($listener)) {
+				$listener->remove();
+				log::add('jMQTT', 'debug', 'Removed Listener on #'.$this->getHumanName().'#');
+			}
+			$this->setCache('listener', null);
+		}
+	}
+
+	public static function listenerAction($_options) {
+		$cmd = self::byId($_options['cmd']);
+		if (!is_object($cmd) || !$cmd->getEqLogic()->getIsEnable() || !$cmd->getType() == 'action' || !$cmd->getConfiguration('auto_publish', 0)) {
+			$listener = listener::byId($_options['listener_id']);
+			$listener->remove();
```

Comment:
better to call listenerUpdate?
less performance but better maintainability

Reply:
It is not possible, as listenerUpdate() is a instance method.
(By example : case where command has been removed)

Reply:
mm, $cmd is instanced just before but you're right. the first case imply $cmd is null so, we won't handle cases differently : counter productive